core/src/main/java/org/jruby/truffle/nodes/rubinius/TimePrimitiveNodes.java

```diff
@@ -0,0 +1,315 @@
+
+        @Specialization
+        public RubyTime timeSSpecific(long seconds, int nanoseconds, Object isGMT, Object offset) {
+            return timeSSpecific(seconds, (long) nanoseconds, isGMT, offset);
+        }
+
+        @Specialization
+        public RubyTime timeSSpecific(int seconds, long nanoseconds, Object isGMT, Object offset) {
+            return timeSSpecific((long) seconds, nanoseconds, isGMT, offset);
+        }
+
+        @Specialization
+        public RubyTime timeSSpecific(long seconds, long nanoseconds, Object isGMT, Object offset) {
+            // TODO(CS): overflow checks here in Rbx
+            final RubyTime time = new RubyTime(getContext().getCoreLibrary().getTimeClass(), seconds, nanoseconds);
+            writeIsGMTNode.execute(time, isGMT);
+            writeOffsetNode.execute(time, offset);
+            return time;
+        }
+
+    }
+
+    @RubiniusPrimitive(name = "time_seconds")
+    public static abstract class TimeSecondsPrimitiveNode extends RubiniusPrimitiveNode {
+
+        public TimeSecondsPrimitiveNode(RubyContext context, SourceSection sourceSection) {
+            super(context, sourceSection);
+        }
+
+        public TimeSecondsPrimitiveNode(TimeSecondsPrimitiveNode prev) {
+            super(prev);
+        }
+
+        @Specialization
+        public long timeSeconds(RubyTime time) {
+            return time.getSeconds();
+        }
+
+    }
+
+    @RubiniusPrimitive(name = "time_useconds")
+    public static abstract class TimeUSecondsPrimitiveNode extends RubiniusPrimitiveNode {
+
+        public TimeUSecondsPrimitiveNode(RubyContext context, SourceSection sourceSection) {
+            super(context, sourceSection);
+        }
+
+        public TimeUSecondsPrimitiveNode(TimeUSecondsPrimitiveNode prev) {
+            super(prev);
+        }
+
+        @Specialization
+        public long timeUSeconds(RubyTime time) {
+            return time.getNanoseconds();
+        }
+
+    }
+
+    @RubiniusPrimitive(name = "time_decompose")
+    public static abstract class TimeDecomposePrimitiveNode extends RubiniusPrimitiveNode {
+
+        @Child protected RubyTimeToDateTimeNode toDateTimeNode;
+
+        public TimeDecomposePrimitiveNode(RubyContext context, SourceSection sourceSection) {
+            super(context, sourceSection);
+            toDateTimeNode = new RubyTimeToDateTimeNode(context, sourceSection);
+        }
+
+        public TimeDecomposePrimitiveNode(TimeDecomposePrimitiveNode prev) {
+            super(prev);
+            toDateTimeNode = prev.toDateTimeNode;
+        }
+
+        @Specialization
+        public RubyArray timeDecompose(VirtualFrame frame, RubyTime time) {
+            final DateTime dateTime = toDateTimeNode.toDateTime(frame, time);
+            final Object[] decomposed = decompose(dateTime);
+            return new RubyArray(getContext().getCoreLibrary().getArrayClass(), decomposed, decomposed.length);
+        }
+
+        @CompilerDirectives.TruffleBoundary
```

Comment:
OK, I thought these would be simple getters. The Joda library looks indeed very powerful with lots of classes and abstractions. Do we have any alternatives or is most of the functionality really required?